src/nanoemoji/write_font2png_html.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Writes an html file suitable to screenshot for a single font glyph"""
+
+
+from absl import app
+from absl import flags
+from nanoemoji import codepoints, util
+from pathlib import Path
+import textwrap
+
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string("output_file", "-", "Output filename ('-' means stdout)")
+flags.DEFINE_integer("resolution", None, "Resolution in px")
+
+
+def main(argv):
+    assert len(argv) == 3
+    font_file = Path(argv[1]).name
+    activation = "".join(
+        f"&#x{cp:04x};" for cp in codepoints.from_filename(Path(argv[2]).name)
+    )
+    with util.file_printer(FLAGS.output_file) as print:
+        print(
+            textwrap.dedent(
+                """
+            <!DOCTYPE html>
+            <style>
+                @font-face {
+                    font-family: "TestFont";
+                    src: url("FONT_LOCATION");
+                }
+                * {
+                    margin: 0;
+                    overflow: hidden;
+                    font-family: "TestFont";
+                }
+            </style>
+            <span id="glyph" style="font-size: 16em;">ACTIVATION</span>
+            <script>
+                let t0 = performance.now();
+                (async function() {
+                  let fontLoads = [];
+                  for (fontFace of document.fonts.values()) {
+                    fontLoads.push(fontFace);
+                  }
+                  await Promise.all(fontLoads);
+                  console.log("Fonts ready.", performance.now() - t0, "ms");
+
+                  let glyph = document.getElementById("glyph");
+
+                  // try to hit target height by adjusting size in em
+                  let currEm = parseInt(glyph.style.fontSize);
+                  let newEm = currEm * 256 / glyph.offsetHeight;
+                  glyph.style.fontSize = `${newEm}em`;
+
+
+                })();
+            </script>
+        """.replace(
+                    "FONT_LOCATION", font_file
```

Comment:
better to use f-string interpolation instead of .replace()

Reply:
It's quite confusing to manage a f-string that is itself a container for js-interpolations like ```glyph.style.fontSize = `${newEm}em`;```, added a comment about that.